lib/chatwithtools.py

```diff
@@ -43,42 +41,49 @@ def try_parse_tool_calls(content: str):
 
 class ChatWithToolsProcessor:
     """
-    A chat with tools processor
-    """
+	A chat with tools processor that supports batch processing
+	"""
 
     model: ChatLlamaCpp
 
     def __init__(self, runner: ChatLlamaCpp):
         self.model = runner
 
-    def __call__(
-            self,
-            inputs: dict[str, Any],
-    ) -> dict[str, str]:
-        model_with_tools = self.model.bind_tools(json.loads(inputs['tools']))
+    def _process_single_input(self, input_data: dict[str, Any]) -> dict[str, Any]:
+        model_with_tools = self.model.bind_tools(json.loads(input_data['tools']))
```

Comment:
the key `tools`'s presence can be checked here, also `json.loads` might throw an exception
A pydantic `BaseModel` for validation would be ideal but just this one check also works since other keys will be present mostly.

Reply:
if it throws we fail the task. if tools does not exist the task will fail as well as it's chat with tools we expect tools to be present